### ISSUE TITLE
ad.quantities.weighted_variance returns weighted standard deviation

### ISSUE THREAD

**Issue by RaymondSimons** [yt-project/yt#2713]
### Bug report

**Bug summary**

The calculation of `ad.quantities.weighted_variance` returns the weighted standard deviation and not the weighted variance. Something similar has been noted in profile.variance, but there is no note in class WeightedVariance.

https://github.com/yt-project/yt/blob/9c802346351b18e25f56f38f9b90a698cec41a6b/yt/data_objects/profiles.py#L91
```python
    def variance(self):
        issue_deprecation_warning("""
profile.variance incorrectly returns the profile standard deviation and has 
been deprecated, use profile.standard_deviation instead."""
        )
        return self.standard_deviation
```

https://github.com/yt-project/yt/blob/9c802346351b18e25f56f38f9b90a698cec41a6b/yt/data_objects/derived_quantities.py#L348
```python
class WeightedVariance(DerivedQuantity):
    r"""
    Calculates the weighted variance and weighted mean for a field
    or list of fields. Returns a YTArray for each field requested; if one,
    it returns a single YTArray, if many, it returns a list of YTArrays
    in order of the listed fields.  The first element of each YTArray is
    the weighted variance, and the second element is the weighted mean.
    Where f is the field, w is the weight, and <f_w> is the weighted mean,
    the weighted variance is
    Sum_i( (f_i - <f_w>)^2 \* w_i ) / Sum_i(w_i).
```

**Code for reproduction**

Using D9p_500 on http://yt-project.org/data

```python
import yt
import numpy as np

ds = yt.load('10MpcBox_HartGal_csf_a0.500.d')
ad = ds.all_data()
var_yt = ad.quantities.weighted_variance(fields = ('stars', 'particle_velocity_x'), weight = ('stars', 'particle_ones'))
std_np = np.std(ad['stars', 'particle_velocity_x'])

print ('quantities.weighted_variance: {}'.format(var_yt[0].to('km/s')))
print ('np.std: {}'.format(std_np.to('km/s')))
```

**Actual outcome**

```
quantities.weighted_variance: 21.237811059938895
np.std: 21.237811059938895
```

**Expected outcome**

Expect the result of quantities.weighted_variance to be equal to np.std**2.

**Version Information**

  * Operating System: Mac Os Mojave
  * Python Version: Python 3.5.6 :: Anaconda, Inc
  * yt version: 3.4.1
  * Other Libraries (if applicable): 

**edit** by @neutrinoceros : added links to source

**Comment by jzuhone** [comment]
@jzuhone I'll address this

**Comment by munkm** [comment]
I just assigned it to you @jzuhone

**Comment by munkm** [comment]
Sorry for taking a while to get to you @RaymondSimons! Thank you so much for the bug report.

**Comment by munkm** [comment]
Closed via yt-project/yt#3132
